simperby/src/node/stf.rs

```diff
@@ -0,0 +1,288 @@
+//! The definition of the state transition function which deduces
+//! the next block from the current state, current block and the next given block.
+
+use super::state_storage::*;
+use crate::*;
+
+#[derive(Error, Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub enum ExecutionError {
+    /// The first failed transaction (later transactions will not be executed).
+    #[error("transaction failed: index({index:}) / {message:}")]
+    TransactionFailure { index: usize, message: String },
+    #[error("state storage error: {0}")]
+    StateStorageError(StateStorageError),
+    #[error("given header is invalid: {0}")]
+    InvalidHeader(String),
+    #[error("given state is invalid: {0}")]
+    InvalidState(String),
+}
+
+impl From<StateStorageError> for ExecutionError {
+    fn from(e: StateStorageError) -> Self {
+        ExecutionError::StateStorageError(e)
+    }
+}
+
+/// Partial data about the `BlockHeader` that is not part of the block execution result.
+/// In other words, the set of information that is needed to fill the next `BlockHeader`
+/// but can't be deduced from the next block itself.
+pub struct HeaderInfo {
+    pub author: PublicKey,
+    pub prev_block_finalization_proof: FinalizationProof,
+    pub timestamp: Timestamp,
+}
+
+/// Executes the block, updating the storage.
+///
+/// Note that it does not interact with the storage checkpoint;
+/// the caller must check the result and
+/// revert the state if it is not successful.
+pub(super) async fn execute_block(
+    storage: &mut StateStorage,
+    last_header: BlockHeader,
+    transactions: Vec<Transaction>,
```

Comment:
```suggestion
    transaction_list: Vec<Transaction>,
```

Reply:
I have consistently used plurals instead of `_list`. Let's keep it.